Sunrise/ViewModels/Categories/CategoriesViewModel.swift

```diff
@@ -0,0 +1,192 @@
+import Commercetools
+
+class CategoriesViewModel: BaseViewModel {
+
+    // Due to https://bugs.swift.org/browse/SR-773
+    typealias Category = Commercetools.Category
+
+    // Inputs
+    let refreshObserver: Observer<Void, NoError>
+    let selectedRowObserver: Observer<IndexPath, NoError>
+
+    // Outputs
+    let rootCategoryNames = MutableProperty([String?]())
+    let activeRootCategoryName: MutableProperty<String?> = MutableProperty(nil)
+    let contentChangesSignal: Signal<Changeset, NoError>
+    let backgroundImage: MutableProperty<UIImage?> = MutableProperty(nil)
+    let title = NSLocalizedString("Categories", comment: "Categories")
+
+    // Actions
+    lazy var selectRootCategoryAction: Action<String, Void, NoError> = { [unowned self] in
+        return Action(enabledIf: Property(value: true), { categoryName in
+            if let rootCategory = self.rootCategories.value.filter({ $0.name?.localizedString == categoryName }).first,
+               self.activeCategories.value.count == 0 ||
+               self.activeCategories.value.first?.name?.localizedString != categoryName {
+                self.activeCategories.value = [rootCategory]
+                self.activeRootCategoryName.value = rootCategory.name?.localizedString
+            }
+            return SignalProducer.empty
+        })
+    }()
+
+    private let contentChangesObserver: Observer<Changeset, NoError>
+
+    private let rootCategories = MutableProperty([Category]())
+    private let activeCategories = MutableProperty([Category]())
+    private let childCategoriesCache = MutableProperty([String: [Category]]())
+
+    // MARK: - Lifecycle
+
+    override init() {
+        let (refreshSignal, observer) = Signal<Void, NoError>.pipe()
+        refreshObserver = observer
+
+        (contentChangesSignal, contentChangesObserver) = Signal<Changeset, NoError>.pipe()
+        let (selectedRowSignal, selectedRowObserver) = Signal<IndexPath, NoError>.pipe()
+        self.selectedRowObserver = selectedRowObserver
+
+        super.init()
+
+        rootCategoryNames <~ rootCategories.producer.map { rootCategories in rootCategories.map({ $0.name?.localizedString }) }
+        // set the first root category to be an active one
+        activeCategories <~ rootCategories.producer.map { rootCategories in rootCategories.first != nil ? [rootCategories.first!] : [] }
+        activeRootCategoryName <~ activeCategories.producer.map { $0.first?.name?.localizedString }
+        activeCategories.combinePrevious(activeCategories.value).signal.observeValues { [weak self] previous, current in
+            self?.updateActiveCategory(from: previous, to: current)
+        }
+        // Use hardcoded images till we get assets on categories
+        backgroundImage <~ activeCategories.producer.map { (activeCategories: [Category]) -> UIImage? in
+            guard let rootCategoryName = activeCategories.first?.name?.localizedString else { return nil }
+            switch rootCategoryName {
```

Comment:
@cneijenhuis - currently we have 3 images, two specific to _Men_ and _Women_ categories, and I'm using the third one for all other categories.